app/models/manageiq/providers/kubernetes/container_manager/metrics_capture.rb

```
@@ -57,7 +58,11 @@ def perf_collect_metrics(interval_name, start_time = nil, end_time = nil)
                 "[#{start_time}] [#{end_time}]")
 
       begin
-        context = CaptureContext.new(target, start_time, end_time, INTERVAL)
+        context = if ems && ems.connection_configurations.prometheus.try(:endpoint)
+                    PrometheusCaptureContext.new(target, start_time, end_time, INTERVAL)
+                  else
```

I agree this feels wrong (seems to assume prometheus or hawkular). 
Can you please check if this code is even called when we have no metric endpoint?
if that is the case (should be the case) you can raise an error unless prometheus OR hawkular